R/phenotypes.R

```diff
@@ -277,6 +277,177 @@ setPheno = function(pop, h2=NULL, H2=NULL, varE=NULL, corE=NULL,
+#' @param p Numeric or \code{NULL}. The p-value used for environmental
+#'   covariates in GxE traits. If \code{NULL}, a value is sampled
+#'   randomly. Passed to \code{\link{setPheno}()} when
+#'   \code{force = TRUE}.
+#' @param onlyPhenoPop Logical. If \code{TRUE} the function returns the
+#'   population-level phenotype matrix (see Details). If \code{FALSE}
+#'   (default) the input \code{pop} object is returned with its
+#'   \code{pop@miscPop$pheno} slot updated.
+#' @param simParam A \code{\link{SimParam}} object. If \code{NULL}, the
+#'   global \code{SP} is used.
+#' @param ... Additional arguments passed to \code{FUN()}.
+#'
+#' @details
+#' For a \code{MultiPop} input, \code{setPhenoPop()} is applied to every
+#' population inside the \code{MultiPop} and each population's result is
+#' stored in it's \code{pop@miscPop$pheno} slot. When a
+#' population already has \code{pop@miscPop$pheno} present, only the
+#' columns specified by \code{traits} are overwritten; other columns are
+#' preserved.
+#'
+#' If \code{force = TRUE}, individual-level phenotypes are created (or
+#' replaced) by calling \code{\link{setPheno}()} using the supplied
+#' \code{h2}, \code{H2}, \code{varE}, \code{corE}, \code{reps}, \code{p}
+#' and \code{fixEff} arguments. See \code{\link{setPheno}} documentation
+#' for details about those arguments and their interpretation.
+#'
+#' The \code{FUN} argument should return one numeric summary value per
+#' trait. The output produced by applying
+#' \code{FUN} to \code{pop@pheno} is transposed and stored as a single-row
+#' matrix with \code{pop@nTraits} columns.
+#'
+#' @return
+#' If \code{onlyPhenoPop = FALSE} (default), the input object
+#' (\code{Pop}, \code{HybridPop} or \code{MultiPop}) is returned with
+#' \code{pop@miscPop$pheno} populated or updated. If
+#' \code{onlyPhenoPop = TRUE}, a numeric matrix is returned:
+#' - for a single \code{Pop}, a 1-by-\code{nTraits} matrix; and
+#' - for a \code{MultiPop}, a \code{length(pop)} (rows) by \code{nTraits}
+#'   (columns) matrix where each row is the population-level summary for
+#'   one constituent population (in the same order as \code{pop@pops}).
+#'
+#' @examples
+#' # Create founder haplotypes
+#' founderPop <- quickHaplo(nInd = 4, nChr = 1, segSites = 10)
+#'
+#' # Set simulation parameters and a single additive trait
+#' SP <- SimParam$new(founderPop)
+#' SP$addTraitA(10)
+#'
+#' # Create a population and a multi-population object
+#' pop <- newPop(founderPop, simParam = SP)
+#' multiPop <- newMultiPop(pop[1:2], pop[3:4])
+#'
+#' # Ensure individual phenotypes exist (error variance = 1)
+#' multiPop <- setPheno(multiPop, varE = 1)
+#'
+#' # Compute population means (one mean per trait per population)
+#' multiPop <- setPhenoPop(multiPop, FUN = colMeans)
+#'
+#' # Compute medians and return only the population-level matrix
+#' medians <- function(x) apply(x, 2, median)
+#' phenoMat <- setPhenoPop(multiPop, onlyPhenoPop = TRUE, FUN = medians)
```

Comment:
```suggestion
#' setPhenoPop(multiPop, onlyPhenoPop = TRUE, FUN = medians)
```

So we print out and users can compare with the above mean